src/ch/imedias/rsccfx/model/Rscc.java

```diff
@@ -154,22 +165,60 @@ public void requestKeyFromServer() {
     keyServerSetup();
 
     String command = commandStringGenerator(
-        pathToResourceDocker, "start_x11vnc.sh", pathToStunDumpFile);
+        pathToResourceDocker, "port_share.sh", vncPort.getValue(), pathToStunDumpFile);
     String key = systemCommander.executeTerminalCommand(command);
     setKey(key); // update key in model
+    startVncServer();
   }
 
   /**
    * Starts connection to the user.
    */
   public void connectToUser() {
     keyServerSetup();
+    String command = commandStringGenerator(pathToResourceDocker,
+        "port_connect.sh", vncPort.getValue(), getKey());
+    systemCommander.executeTerminalCommand(command);
+    startVncViewer("localhost");
+  }
+
+  /**
+   * Starts the VNC Server.
+   */
+  public void startVncServer() {
+    StringBuilder vncServerAttributes = new StringBuilder("-bg -nopw -q -localhost");
```

Comment:
Yes, it restricts the server to only accept connections from this interface (equal as 127.0.0.1) and in every scenario the requests comes from there